### DISCUSSION TITLE
Rethink build/serve for Tenant UI frontend

### DISCUSSION THREAD

**Discussion by loneil** [bcgov/traction#628] [Ideas]
TBD

The Tenant UI was developed to :

- Be served by it's Node API. This Node API also supplies some endpoints that the Tenant UI needs (SMTP functionality, OIDC login for Innkeeper, etc).
- Get it's env config from the Node API
- Build... (refresh memory here, can't remember what was thrown together originally). The Vue app is built as a static dist and that is served by the Node api on an express route.
- Startup... (same as above)

Some of this could be rethought if necessary, as the Vue app is pretty standalone at this point.

**Comment by loneil** [comment]
The FE (Vue app) could be served by some other structure (Caddy, Nginx) and the node API continue on it's own.
Not sure if that's better to have 2 servers to host and maintain rather than just having the Node app handle both purposes but think it could work.

**Comment by amanji** [comment]
My general developer experience is I find the nested structure confusing to understand from an architectural perspective. Many of the build configs have the Vue frontend build hidden within the API build (which caused a couple of issues for me not being initially aware of this). I also ran into issues with using a different test framework for the Vue frontend which required some additional configuration.

My thoughts (and this is strictly my own opinion) is that I favour having clear boundaries and explicit separation between the API and the Vue frontend (unless there is some clear reason I've overlooked here). I think this also makes sense from a scalability perspective to have both separate. I do however see a reason to keep the nested structure for those that don't want to now deal with deploying multiple services.

**Comment by WadeBarnes** [comment]
A pattern we've used in several other projects it to serve out the FE on a pod that also acts as a proxy server to the backend API.  This way you have a single URL for the application and from the outside it looks like a single app.  Example; https://github.com/bcgov/orgbook-bc-client

This allows you to resource and scale the services separately as needed.
